pkg/services/folder/folderimpl/folder_unifiedstorage.go

```diff
@@ -155,6 +167,61 @@ func (s *Service) getFromApiServer(ctx context.Context, q *folder.GetFolderQuery
 	return f, err
 }
 
+func (s *Service) getFolderByIDFromApiServer(ctx context.Context, id int64, orgID int64) (*folder.Folder, error) {
+	if id == 0 {
+		return &folder.GeneralFolder, nil
+	}
+
+	folderkey := &resource.ResourceKey{
+		Namespace: s.k8sclient.getNamespace(orgID),
+		Group:     "folder.grafana.app",
+		Resource:  "folders",
```

Comment:
Small nit, could we use the constants instead?
```suggestion
		Group:     v0alpha1.FolderResourceInfo.GroupVersionResource().Group,
		Resource: v0alpha1.FolderResourceInfo.GroupVersionResource().Resource,
```

Reply:
oh!!! good call!